CHANGELOG.md

```diff
@@ -1,5 +1,113 @@
+#### :gear: User roles preserve the local storage ([#1454](https://github.com/DevExpress/testcafe/issues/1454))
+
+TestCafe now saves the local storage state when switching between roles, so that when you switch back you get the same local storage content you left.
+
+This is useful for testing websites that perform authentication via local storage instead of cookies.
+
+#### Internal Improvements
+
+* `StorageSandbox` is shared to the client API with `clear` and `lock` methods added. ([testcafe-hammerhead/#1189](https://github.com/DevExpress/testcafe-hammerhead/issues/1189))
+* The number of failed cookie-sync messages is now limited. ([testcafe-hammerhead/#1193](https://github.com/DevExpress/testcafe-hammerhead/issues/1193))
+* `Hammerhead.backupStorages` and `Hammerhead.restoreStorages` methods are added to the client API([testcafe-hammerhead/#1125](https://github.com/DevExpress/testcafe-hammerhead/issues/1125))
+* The `GetEventListeners` method is implemented in `sandbox.event.listeners`. ([testcafe-hammerhead/#1238](https://github.com/DevExpress/testcafe-hammerhead/issues/1238))
+* JavaScript expressions now supported in raw API selectors. ([#1466](https://github.com/DevExpress/testcafe/issues/1466))
+
+### Bug Fixes
+
+* Selector's `withAttribute` method supports search by strict match ([#1548](https://github.com/DevExpress/testcafe/issues/1548]))
+* Description for the `path` parameter of the `t.takeScreenshot` action has been corrected ([#1515](https://github.com/DevExpress/testcafe/issues/1515))
+* Local storage is now cleaned appropriately after the test run.([#1546](https://github.com/DevExpress/testcafe/issues/1546))
+* TestCafe now checks element visibility with a timeout when the target element's `style.top` is negative ([#1185](https://github.com/DevExpress/testcafe/issues/1185))
+* Fetching an absolute CORS URL now works correctly. ([#1629](https://github.com/DevExpress/testcafe/issues/1629))
+* Add partial support for proxying live node collections (the `GetElementsByTagName` method) ([#1442](https://github.com/DevExpress/testcafe/issues/1442))
+* TypeScript performance has been enhanced. ([#1591](https://github.com/DevExpress/testcafe/issues/1591))
+* The right port is now applied to a cross-domain iframe location after redirect. ([testcafe-hammerhead/#1191](https://github.com/DevExpress/testcafe-hammerhead/issues/1191))
+* All internal properties are marked as non-enumerable. ([testcafe-hammerhead/#1182](https://github.com/DevExpress/testcafe-hammerhead/issues/1182))
+* Support proxying pages with defined referrer policy. ([testcafe-hammerhead/#1195](https://github.com/DevExpress/testcafe-hammerhead/issues/1195))
+* WebWorker content is now correctly proxied in FireFox 54. ([testcafe-hammerhead/#1216](https://github.com/DevExpress/testcafe-hammerhead/issues/1216))
+* Methods that return live node collections are now correctly proxied. ([testcafe-hammerhead/#1096](https://github.com/DevExpress/testcafe-hammerhead/issues/1096))
```

Comment:
Drop this.
It is dublicate of the `Add partial support for proxying live node collections (the `GetElementsByTagName` method) ([#1442](https://github.com/DevExpress/testcafe/issues/1442))`